pkg/util/utils_linux.go

```diff
@@ -83,8 +83,14 @@ func isVirtualConsoleDevice(device string) bool {
 			  increase the startup time needlessly or made the code more complex
 			  than needed.
 	*/
-	matched, _ := path.Match("/dev/tty[0-9]*", device)
-	return matched
+	suffix := strings.TrimPrefix(path, "/dev/tty")
```

Comment:
Might be good to do a filepath.Clean() on path first, to make sure duplicated `/` characters and other things that could confuse this get removed.

Reply:
IMHO, given how seldomly this function is gonna be called, it doesn't hurt to be defensive and document/enforce expectations as codebases do evolve with time.

Reply:
With the defensive argument, we could add filepath.Clean() calls to every function where a path is passed. Despite the caller is providing a clean path. That would be too bloated and redundant.

If the code changes and suddenly an unclean path would be passed, it could affect also other instances were the caller (`AddPrivilegedDevices`) and it's callers pass the uncleaned path. Instead of having to clean the path multiple times in all places were `path` is passed to, it would be better to do it only 1x at the source.
In the caller function unclean paths could for example also cause that `d.Path == "/dev/ptmx"` would not match anymore, despite being the path to the ptmx device.
Because this is also a private function that is called in 1 function. I don't see the risk here.

What do you think of adding a godoc comment instead, documenting that it is needed.

Reply:
> With the defensive argument, that code can change, we could reason to add filepath.Clean() calls to every function where a path is passed. Despite that the caller is providing a clean path already. That would get very bloated and redundant.

Not every function is matching device path though, but I hear you!

> If this would be an exported function I would totally agree because we would not know what the caller provides.
But this is an internal function only used by 1 caller with known controlled input.

That is indeed a good point.

> What do you think of adding a godoc comment instead, documenting that a clean path is expected?

Sounds like an acceptable compromise to me :) Thanks!

Reply:
@mheon @TomSweeneyRedHat would that also be fine for you? Or am I missing a reason to clean the path in `isVirtualConsoleDevice`?

Reply:
I generally just program defensively by default, but I won't argue the point too much - seems fine to me.

Reply:
I added a godoc comment for the function, removed the obsolete comment about "path.Match" and rebased the branch.

Reply:
/approve

Reply:
For completeness we should add a check for
if !HasPrefix(path, "/dev/tty") {
    return false
}

Reply:
otherwise a path == 1234 would pass.

Reply:
The 
> If s doesn't start with prefix, s is returned unchanged. 

property of `strings.TrimPrefix` is used currently (https://pkg.go.dev/strings#TrimPrefix).

if path does not have the prefix `strings.TrimPrefix` returns the unchanged path, `suffix == path` will match and false will be returned.

Reply:
Sure, but "1234" is not a DEVICE.

Reply:
The current code returns false if the input path is "1234".
Proof: https://go.dev/play/p/EyxU04tIUug
This happens because of:
```go
	suffix := strings.TrimPrefix(path, "/dev/tty")
	if suffix == path || suffix == "" {
		return false
```

If `path` is `1234` `suffix == path` matches and `false` is returned.

Reply:
I added more unit-testcases, amonst others one for `path == "1234`:
https://github.com/containers/podman/pull/17104/files#diff-251941df57512b4f663415d5e3d02fe53323335027a62d68e363ffa4195d5fe9R29

Reply:
Ok I missed that.